### PR TITLE
feat: DW Openpose Processor

### PR DESCRIPTION
**Pull request by blessedcoolant**
## What type of PR is this? (check all applicable)

- [x] Feature

## Have you discussed this change with the InvokeAI team?
- [x] Yes

      
## Have you updated all relevant documentation?
- [x] No

## Description

This PR adds the DW Openpose Image Processor for ControlNet to InvokeAI.

- Added a new node: `DW Openpose Image Processor` - This node takes in an image and lets you generate an openpose skeleton from it. You are able to generate the body skeleton, hand skeleton and also the facial landmarks selectively.
- DW Openpose Processor has been implemented in the Linear UI. It is also now the default processor for Openpose.

## Removed

The old `Openpose Image Processor` has been entirely removed. Any current workflow using this node will break. But the fix is simple. Just replace it with the new `DW Openpose Image Processor` and you are good to go.

## After Merge

- Update the docs to reflect this change

### CODE REVIEW

File: invokeai/app/invocations/controlnet_image_processors.py [hunk 1]
```diff
@@ -633,3 +634,30 @@ def run_processor(self, image):
...
+)
+class DWPoseImageProcessorInvocation(ImageProcessorInvocation):
+    """Generates an openpose pose from an image using DWPose"""
+
+    draw_body: bool = InputField(default=True)
+    draw_face: bool = InputField(default=False)
+    draw_hands: bool = InputField(default=False)
+    image_resolution: int = InputField(default=512, ge=0, description=FieldDescriptions.image_res)
+
+    def run_processor(self, image):
```
Comment (psychedelicious) [inline invokeai/app/invocations/controlnet_image_processors.py:654]: types pls

Comment (blessedcoolant) [reply invokeai/app/invocations/controlnet_image_processors.py:654]: types for?

File: invokeai/backend/image_util/dwpose/onnxpose.py
Comment (psychedelicious) [inline invokeai/backend/image_util/dwpose/onnxpose.py:1]: Where did this code come from and what is the license?

Comment (blessedcoolant) [reply invokeai/backend/image_util/dwpose/onnxpose.py:1]: The original `DWPose` implemention. Apache 2.0 https://github.com/IDEA-Research/DWPose

**Review by psychedelicious — CHANGES_REQUESTED** [review]
Looks good just would like to have types on our code, and attribution for code that was copied

File: invokeai/app/invocations/controlnet_image_processors.py (hunk 1 above)
Comment (psychedelicious) [reply invokeai/app/invocations/controlnet_image_processors.py:654]: So long as when I edit it I get hints in IDE 👍

File: invokeai/backend/image_util/dwpose/onnxpose.py
Comment (psychedelicious) [reply invokeai/backend/image_util/dwpose/onnxpose.py:1]: Ok let's add a comment to the init or somewhere that links to it

File: invokeai/app/invocations/controlnet_image_processors.py (hunk 1 above)
Comment (blessedcoolant) [reply invokeai/app/invocations/controlnet_image_processors.py:654]: It does.

File: invokeai/backend/image_util/dwpose/onnxpose.py
Comment (blessedcoolant) [reply invokeai/backend/image_util/dwpose/onnxpose.py:1]: Done

**Review by psychedelicious — APPROVED** [review]

**Comment by Millu** [comment]
Should we default the processor to hands as well?

**Comment by hipsterusername** [comment]
works for me. have we had a test on mac?

**Review by hipsterusername — APPROVED** [review]

**Comment by blessedcoolant** [comment]
@Millu Millun, could you update (or point out to me) the docs that need to reflect the fact that the Openpose Image Processor is now removed entirely and the new way to do Openpose is with the `DW Openpose Processor` and that this will break older workflows that use the other processor.

**Review by Millu — APPROVED** [review]
Updated docs & tested on macOS

**Comment by blessedcoolant** [comment]
@hipsterusername Good to merge I think.

**Comment by Millu** [comment]
Let's hold to merge until 3.6.3 is released as this will introduce a breaking change

**Pull request merged**